kansha/board/comp.py

```diff
@@ -325,12 +324,8 @@ def update_card_position(self, data):
             dest = cols[data['dest']]
             card = orig.remove_card(data['card'])
             dest.insert_card(card, data['index'])
-            values = {'from': orig.title().text,
-                      'to': dest.title().text,
-                      'card': card().data.title}
-            notifications.add_history(self.data, card().data,
-                                      security.get_user().data,
-                                      u'card_move', values)
+            event = ('card_move', [card(), dest], orig)
+            notifications.add_event_to_history(self, event)
```

Comment:
At least, every component must have a `emit_event(type, data)` function which initialize the sources with `self`